src/rational/mat_q/sample/gauss.rs

```diff
@@ -0,0 +1,172 @@
+
+//! This module contains sampling algorithms for Gaussian distributions over [`MatQ`].
+
+use std::fmt::Display;
+
+use crate::{
+    error::MathError,
+    rational::{MatQ, Q},
+    traits::{GetEntry, GetNumColumns, GetNumRows, SetEntry},
+};
+
+impl MatQ {
+    /// Chooses a [`MatQ`] instance according to the continuous Gaussian distribution.
+    /// Here, each entry is chosen according to the provided distribution.
+    ///
+    /// Parameters:
+    /// - `center`: specifies the center for each entry of the matrix individually
+    /// - `sigma`: specifies the standard deviation
+    ///
+    /// Returns new [`MatQ`] sample chosen according to the specified continuous Gaussian
+    /// distribution or a [`MathError`] if the specified parameters were not chosen
+    /// appropriately (`sigma > 0`).
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::rational::MatQ;
+    ///
+    /// let sample = MatQ::sample_gauss(&MatQ::new(5, 5), 1).unwrap();
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type [`NonPositive`](MathError::NonPositive)
+    ///     if `sigma <= 0`.
+    pub fn sample_gauss(center: &MatQ, sigma: impl Into<f64>) -> Result<MatQ, MathError> {
+        let mut out = MatQ::new(center.get_num_rows(), center.get_num_columns());
+        let sigma = sigma.into();
+
+        for i in 0..out.get_num_rows() {
+            for j in 0..out.get_num_columns() {
+                let center_entry_ij = center.get_entry(i, j)?;
+                let sample = Q::sample_gauss(center_entry_ij, sigma)?;
+                out.set_entry(i, j, sample)?
+            }
+        }
+
+        Ok(out)
+    }
+
+    /// Chooses a [`MatQ`] instance according to the continuous Gaussian distribution.
+    /// Here, each entry is chosen according to the provided distribution and each entry
+    /// is sampled with the same center.
+    ///
+    /// Parameters:
+    /// - `num_rows`: specifies the number of rows of the sampled matrix
+    /// - `num_cols`: specifies the number of columns of the sampled matrix
+    /// - `center`: specifies the same center for each entry of the matrix
+    /// - `sigma`: specifies the standard deviation
+    ///
+    /// Returns new [`MatQ`] sample chosen according to the specified continuous Gaussian
+    /// distribution or a [`MathError`] if the specified parameters were not chosen
+    /// appropriately (`sigma > 0`).
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::rational::{Q, MatQ};
+    ///
+    /// let center = Q::from((5,2));
+    ///
+    /// let sample = MatQ::sample_gauss_same_center(5, 5, &center, 1).unwrap();
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type [`NonPositive`](MathError::NonPositive)
+    ///     if `sigma <= 0`.
+    ///
+    /// # Panics ...
+    /// - if the number of rows or columns is negative, `0`, or does not fit into an [`i64`].
+    pub fn sample_gauss_same_center(
```

Comment:
```suggestion
    pub fn sample_gauss(
```

Reply:
talked about it with Laurens when I made the naming this way -> will leave it as it is